src/Platform/AcquiaCloudPlatform.php

```diff
@@ -182,21 +192,38 @@ public static function getEnvironmentQuestion(Config $config, AcquiaCloudClientF
    * {@inheritdoc}
    */
   public function execute(Command $command, InputInterface $input, OutputInterface $output) : int {
-    $environments = new Environments($this->getAceClient());
     $sites = $this->getPlatformSites();
     if (!$sites) {
       $output->writeln('<warning>No sites available. Exiting...</warning>');
       return 1;
     }
 
+    $group_name = $input->getOption('group');
     $input_uri = $input->getOption('uri');
+
+    if ($group_name && $input_uri) {
+      $helper = $command->getHelper('question');
+      $question = new ConfirmationQuestion('You have provided both the options, group as well as uri. We will ignore the uri option. Do you want to proceed (y/n)?', TRUE);
+      if (!$helper->ask($input, $output, $question)) {
+        return 2;
+      }
+    }
+
+    if ($input->hasOption('group') && $group_name) {
+      $sites = $this->filterSitesByGroup($group_name, $sites, $output);
+      if (empty($sites)) {
+        return 3;
+      }
+    }
+
     $sites = array_column($sites, 'uri');
     $args = $this->dispatchPlatformArgumentInjectionEvent($input, $sites, $command);
     $exit_code = 0;
     $vendor_paths = $this->get(self::ACE_VENDOR_PATHS);
+    $environments = new Environments($this->getAceClient());
     foreach ($this->get(self::ACE_ENVIRONMENT_DETAILS) as $application_id => $environment_id) {
```

Comment:
It won't work if we only filtering the values in `$sites` variable, cause the foreach running through the environment ids. We need to check that the given environment domain has in the sites or not and running the command or just call `continue` accordingly.